### PR TITLE
Add `flow migrate state` command

### LINKED ISSUE #1433: Emulator command for migrating contracts/state to Cadence 1.0
### Issue To Be Solved

The CLI should have a command to take in an Emulator snapshot created using a recent Emulator version with Cadence v0.42, run the Cadence 1.0 contract and state migrations on it, and produce a new Emulator snapshot for the latest version  of the Emulator with Cadence v1.0.

### Suggest A Solution

- Add a command like `flow migrate`
- Call the Emulator's Cadence 1.0 migration function, added in https://github.com/onflow/flow-emulator/pull/557

### PR DESCRIPTION
**Pull request by jribbink**
Closes #1433

## Description

______

For contributor use:

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-cli/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: internal/migrate/state.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve staged contracts: %w", err)
+	}
+
+	store, err := sqlite.New(stateFlags.DBPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open database: %w", err)
+	}
+
+	rwf := &migration.NOOPReportWriterFactory{}
```
Comment (turbolent) [inline internal/migrate/state.go:77]: We'll probably want some kind of writer here to provide feedback to the developer

File: internal/migrate/state.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	for i, contractName := range contractNames {
+		// First try to get contract address from aliases
+		contract, err := state.Contracts().ByName(contractName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get contract by name: %w", err)
+		}
+
+		var address flow.Address
+		if contract.Aliases.ByNetwork(config.EmulatorNetwork.Name) != nil {
+			address = contract.Aliases.ByNetwork(config.EmulatorNetwork.Name).Address
```
Comment (turbolent) [inline internal/migrate/state.go:99]:
```suggestion
		alias := contract.Aliases.ByNetwork(config.EmulatorNetwork.Name)
		if alias != nil {
			address = alias.Address
```

File: internal/migrate/state.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+func migrateState(
+	args []string,
+	globalFlags command.GlobalFlags,
+	_ output.Logger,
+	flow flowkit.Services,
+	state *flowkit.State,
+) (command.Result, error) {
+	if globalFlags.Network != "emulator" {
```
Comment (turbolent) [inline internal/migrate/state.go:63]: Is there maybe a constant defined somewhere that could be used here?

**Review by turbolent — APPROVED** [review]
Nice!

File: internal/migrate/state.go (hunk 3 above)
Comment (turbolent) [reply internal/migrate/state.go]: Addressed in [276d541](https://github.com/onflow/flow-cli/pull/1426/commits/276d5418649f8aa9b3c5b130145a330472173e9f)

File: internal/migrate/state.go (hunk 2 above)
Comment (turbolent) [reply internal/migrate/state.go]: Addressed in [276d541](https://github.com/onflow/flow-cli/pull/1426/commits/276d5418649f8aa9b3c5b130145a330472173e9f)

File: internal/migrate/state.go (hunk 1 above)
Comment (jribbink) [reply internal/migrate/state.go:77]: Sounds good, I'll see how I can surface it.

File: internal/migrate/state.go [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+	"github.com/onflow/flowkit/v2/output"
+	"github.com/rs/zerolog"
+	"github.com/spf13/cobra"
+
+	"github.com/onflow/flow-cli/internal/command"
+)
+
+var stateFlags struct {
+	Contracts  []string `default:"" flag:"contracts" info:"contract names to migrate"`
+	SaveReport string   `default:"" flag:"save-report" info:"save migration report to a given directory if provided"`
```
Comment (jribbink) [inline internal/migrate/state.go:43]: Added a `--save-report` flag - open to feedback here about naming/acceptable defaults.  Assumed it's probably best not to pollute user's workspace if not manually specified (so default to NOOP) but can change if wanted.

cc @turbolent

File: internal/migrate/state.go (hunk 1 above)
Comment (jribbink) [reply internal/migrate/state.go]: I added a `--save-report` flag - open to feedback here about naming/acceptable defaults. Assumed it's probably best not to pollute user's workspace if not manually specified (so default to NOOP) but can change if wanted.

https://github.com/onflow/flow-cli/pull/1426/files#diff-fbaffef2f65ec644d807fb85832bf2077233decf77b14155006c99f7159cd850R43

**Review by SupunS — APPROVED** [review]
Tested with a local build, looks good! 👌

**Review by chasefleming — APPROVED** [review]

**Comment by turbolent** [comment]
Should be good to go once #1450 is in

**Comment by turbolent** [comment]
@SupunS Not quite sure about [5b2daca](https://github.com/onflow/flow-cli/pull/1426/commits/5b2dacac6dea4cf08bf032da58d19027137988f0).

In a previous Emulator state, the Burner contract is not deployed yet, so I think deploying it is correct (instead of updating it).

What about the EVM contract though?

**Comment by SupunS** [comment]
Think that should be OK. We can have the same params as in https://github.com/onflow/flow-emulator/pull/608/files#diff-717939af182d8688c65dd7e4a5b5471360a4e3719f2dba0b34629167f3ce70afR97-R104.

**Pull request merged**